server/instance_cloud_oauth.go

```diff
@@ -0,0 +1,209 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	jira "github.com/andygrunwald/go-jira"
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+
+	"github.com/mattermost/mattermost-plugin-jira/server/utils"
+	"github.com/mattermost/mattermost-plugin-jira/server/utils/types"
+)
+
+type cloudOAuthInstance struct {
+	*InstanceCommon
+
+	// The SiteURL may change as we go, so we store the PluginKey when as it was installed
+	MattermostKey string
+
+	JiraResourceID   string
+	JiraClientID     string
+	JiraClientSecret string
+	JiraBaseURL      string
+}
+
+type CloudOAuthConfigure struct {
+	InstanceURL  string `json:"instance_url"`
+	ClientID     string `json:"client_id"`
+	ClientSecret string `json:"client_secret"`
+}
+
+type JiraAccessibleResources []struct {
+	ID string
+}
+
+var _ Instance = (*cloudOAuthInstance)(nil)
+
+const (
+	JiraScopes        = "read:jira-user,read:jira-work,write:jira-work"
+	JiraScopesOffline = JiraScopes + ",offline_access"
+	JiraResponseType  = "code"
+	JiraConsent       = "consent"
+)
+
+func (p *Plugin) installCloudOAuthInstance(rawURL string, clientID string, clientSecret string) (string, *cloudOAuthInstance, error) {
```

Comment:
Check this everywhere
```suggestion
func (p *Plugin) installCloudOAuthInstance(rawURL , clientID , clientSecret string) (string, *cloudOAuthInstance, error) {
```